internal/model/date_test.go

```diff
@@ -0,0 +1,175 @@
+package model
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"strconv"
+	"testing"
+	"time"
+)
+
+type testJsonDateStruct struct {
+	TestDate Date `json:"testDate"`
+}
+
+func TestNewDate(t *testing.T) {
+	want := Date{Time: time.Date(2020, 12, 31, 0, 0, 0, 0, time.UTC)}
+	assert.Equal(t, want, NewDate("31/12/2020"))
+
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("NewDate should panic with incorrect date format")
+		}
+	}()
+	NewDate("12/31/2020") // wrong format should trigger a panic
```

Comment:
If this is something that can never happen (as per your earlier comment), there shouldn't be a test for it. If it can happen, we shouldn't panic, and should either return an error, or ignore it or return some default.

Reply:
done